dev/scalastyle

```diff
@@ -17,18 +17,10 @@
 # limitations under the License.
 #
 
-SPARK_PROFILES=${1:-"-Pmesos -Pkubernetes -Pyarn -Pspark-ganglia-lgpl -Pkinesis-asl -Phive-thriftserver -Phive"}
+SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
 
-# NOTE: echo "q" is needed because SBT prompts the user for input on encountering a build file
-# with failure (either resolution or compilation); the "q" makes SBT quit.
-ERRORS=$(echo -e "q\n" \
-    | build/sbt \
-        ${SPARK_PROFILES} \
-        -Pdocker-integration-tests \
-        -Pkubernetes-integration-tests \
-        scalastyle test:scalastyle \
-    | awk '{if($1~/error/)print}' \
-)
+SPARK_PROFILES=${1:-"-Pmesos -Pkubernetes -Pyarn -Pspark-ganglia-lgpl -Pkinesis-asl -Phive-thriftserver -Phive -Pdocker-integration-tests -Pkubernetes-integration-tests"}
+ERRORS=$($SCRIPT_DIR/../build/mvn $SPARK_PROFILES scalastyle:check | grep "^error file")
```

Comment:
So, I guess this is a temp fix and we should fix SBT build to use HTTPs correctly .. am I correct?

If this was a non-temp fix, we should rather have another script like `lint-java` vs `sbt-checkstyle`. Ideally we should use SBT in PR builder to save times.

Reply:
? This is a permanent fix for `GitHub Action` and this is not for PR Builder.
For SBT, if you want, you can make a PR~

Reply:
The problem is it affects PR builder:

https://github.com/apache/spark/blob/32af7004a254107340df908e8dd91a0bc22e068e/dev/run-tests.py#L182-L187

and switching to Maven wasn't encouraged; so I had to add such script to use SBT `sbt-checkstyle`.

Reply:
I would like to make a fix for SBT but wasn't able to reproduce. I removed all `.m2` and `.ivy2` but it still works and uses `[info] downloading https://repo1.maven.org/maven2`.

If I am not wrong SBT uses HTTPs:

https://github.com/apache/spark/blob/a3357dfccacadfb40ab103ba1ff3c0927e806dd2/project/SparkBuild.scala#L227

https://www.scala-sbt.org/0.13/docs/Resolvers.html

So, I was wondering how to reproduce.

Reply:
Why don't we simply copy this new file as `mvn-scalastyle` and old file as `scalastyle` as are?

And, Github Actions calls `mvn-scalastyle` and Jenkins PR builder calls `scalastyle`.

FYI, currently Jenkins PR builder directly calls https://github.com/apache/spark/blob/32af7004a254107340df908e8dd91a0bc22e068e/dev/run-tests.py#L190-L196 for Java style instead of `lint-java` for similar reason.

Reply:
I'm +1 for any suggestion if you make a PR and pass all system (including GitHub Action). :)

BTW, did you try to checkout one of older commits than this, @HyukjinKwon ?

If it works in your environment even in that case, it sounds weird to me because of the following.

As you can see on the mailing list, this is reported by @tgravescs and I confirmed the situation. Then, I made this PR at that time. Also, GitHub Action log still shows the whole situation (the **failures** before this commit and the **successes** after this commit).

<img width="593" alt="Screen Shot 2020-01-20 at 11 48 50 PM" src="https://user-images.githubusercontent.com/9700541/72785186-7b43ad00-3bdf-11ea-9ca4-084b7a4b8a58.png">

For now, I don't have any clue about the difference (code side? or Maven server side? or your environment?).

In any way, you can move forward whiling keeping our system **green**. We had better discuss on your PR.

Reply:
I checked this. It still fails before this commit. Please try in your environment, @HyukjinKwon .
```
[error] [FATAL] Non-resolvable parent POM: 
Could not transfer artifact org.apache:apache:pom:18 from/to central (
http://repo.maven.apache.org/maven2): Failed to look for file:
http://repo.maven.apache.org/maven2/org/apache/apache/18/apache-18.pom. Return code is: 501 and 'parent.relativePath' points at wrong local POM @ line 22, column 11
```

Reply:
Thanks for pointer, @dongjoon-hyun!